okta-idx-kotlin/src/main/java/com/okta/idx/kotlin/dto/IdxCapability.kt

```diff
@@ -181,6 +182,56 @@ class IdxTotpCapability internal constructor(
     }
 }
 
+/**
+ * Describes the WebAuthn registration capability associated with an [IdxAuthenticator].
+ *
+ * @property activationData The JSON string containing the public key credential creation options for WebAuthn activation.
+ */
+class IdxWebAuthnRegistrationCapability internal constructor(
+    private val activationData: String,
+) : IdxAuthenticator.Capability {
+
+    /**
+     * Returns the public key credential creation options as a JSON string.
+     *
+     * @param rpId Optional relying party ID. If provided and not blank, overrides the default `rp.id` in the activation data if present.
+     * @return A [Result] containing the JSON string with the `rp.id` field set to [rpId] if specified, or the original activation data otherwise.
+     */
+    fun publicKeyCredentialCreationOptions(rpId: String? = null): Result<String> = runCatching {
+        return if (rpId?.isNotBlank() == true) {
+            val json = JSONObject(activationData).apply {
+                getJSONObject("rp").apply {
+                    put("id", rpId)
+                }
+            }.toString()
```

Comment:
[The workaround I made in the Swift SDK](https://github.com/okta/okta-mobile-swift/pull/236/files#diff-a5efaa48f133b38085c09a4f57756f55f8a14340e04f2c7d7a13c265c196aa81) could be applicable here. The problem I found is that if the rpId doesn't match what the server expects it'll fail anyway.  So my implementation simply infers the rpID from the host value in the`u2fParams.appid`, `extensions.appid`, or issuer URL.